client/solve.go

```diff
@@ -115,6 +115,12 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 	}
 
 	var ex ExportEntry
+	if len(opt.Exports) > 1 {
+		return nil, errors.New("currently only single Exports can be specified")
+	}
+	if len(opt.Exports) == 1 {
+		ex = opt.Exports[0]
+	}
```

Comment:
The eventual call to `Solve` uses `ex` which isn't set if `opt.SessionPreInitialized == true`. Assigning `ex = opt.Exports[0]` shouldn't be related to whether the session is initialized yet or not.